servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletCookies.java

```diff
@@ -38,7 +42,11 @@
      * @param cookies The cookies
      */
     public DefaultServletCookies(javax.servlet.http.Cookie[] cookies) {
-        this.cookies = cookies;
+        if (cookies == null) {
+            this.cookies = new javax.servlet.http.Cookie[]{};
```

Comment:
This should be `this.cookies = new javax.servlet.http.Cookie[0]` to avoid the array initializer and be extracted to a constant

Reply:
Done